cmd/http/close.go

```diff
@@ -32,12 +32,13 @@ func DrainBody(respBody io.ReadCloser) {
 	// If resp.Body is not closed, the Client's underlying RoundTripper
 	// (typically Transport) may not be able to re-use a persistent TCP
 	// connection to the server for a subsequent "keep-alive" request.
+	const maxSlurpSize = 2 << 10
 	if respBody != nil {
 		// Drain any remaining Body and then close the connection.
 		// Without this closing connection would disallow re-using
 		// the same connection for future uses.
 		//  - http://stackoverflow.com/a/17961593/4465767
 		defer respBody.Close()
-		io.Copy(ioutil.Discard, respBody)
+		io.CopyN(ioutil.Discard, respBody, maxSlurpSize)
```

Comment:
Why this limit? AFAICT it will just leave a lot more connections to not be reusable.

It also makes it impossible for it to use the `io.ReaderFrom` interface which is cheaper than the regular copy. 

So I propose to take this out.

Reply:
I borrowed it from Go @klauspost as they have used it as a preferable value.

I can take it out as well.

Reply:
@harshavardhana Yeah, let's for now. They are probably conservative for slow-lorris DOS.